bundles/org.openhab.binding.tivo/src/main/java/org/openhab/binding/tivo/internal/handler/TiVoHandler.java

```diff
@@ -0,0 +1,372 @@
+                }
+            });
+        }
+    }
+
+    /**
+     * {@link chChannelChange} performs channel changing operations.
+     *
+     * @param commandKeyword the TiVo command object.
+     * @param command the command parameter.
+     * @return TivoStatusData status of the command.
+     * @throws InterruptedException
+     */
+    private TivoStatusData chChannelChange(String commandKeyword, String command) throws InterruptedException {
+        int channel = -1;
+        int subChannel = -1;
+
+        TivoStatusData tmpStatus = tivoConnection.get().getServiceStatus();
+        try {
+            // Parse the channel number and if there is a decimal, the sub-channel number (OTA channels)
+            Matcher matcher = NUMERIC_PATTERN.matcher(command);
+            if (matcher.find()) {
+                if (matcher.groupCount() >= 1) {
+                    channel = Integer.parseInt(matcher.group(1).trim());
+                }
+                if (matcher.groupCount() >= 2 && matcher.group(2) != null) {
+                    subChannel = Integer.parseInt(matcher.group(2).trim());
+                }
+            } else {
+                // The command string was not a number, throw exception to catch & log below
+                throw new NumberFormatException();
+            }
+
+            String tmpCommand = commandKeyword + " " + channel + ((subChannel != -1) ? (" " + subChannel) : "");
+            logger.debug("chChannelChange '{}' sending command to tivo: '{}'", getThing().getUID(), tmpCommand);
+
+            // Attempt to execute the command on the tivo
+            tivoConnection.get().cmdTivoSend(tmpCommand);
+            TimeUnit.MILLISECONDS.sleep(tivoConfigData.getCmdWaitInterval() * 2);
+
+            tmpStatus = tivoConnection.get().getServiceStatus();
+
+            // Check to see if the command was successful
+            if (tmpStatus.getConnectionStatus() != ConnectionStatus.INIT && tmpStatus.isCmdOk()) {
+                if (tmpStatus.getMsg().contains("CH_STATUS")) {
+                    return tmpStatus;
+                }
+            } else if (tmpStatus.getConnectionStatus() != ConnectionStatus.INIT) {
+                logger.warn("TiVo'{}' set channel command failed '{}' with msg '{}'", getThing().getUID(), tmpCommand,
+                        tmpStatus.getMsg());
+                switch (tmpStatus.getMsg()) {
+                    case "CH_FAILED NO_LIVE":
+                        tmpStatus.setChannelNum(channel);
+                        tmpStatus.setSubChannelNum(subChannel);
+                        return tmpStatus;
+                    case "CH_FAILED RECORDING":
+                        return tmpStatus;
+                    case "CH_FAILED MISSING_CHANNEL":
+                        return tmpStatus;
+                    case "CH_FAILED MALFORMED_CHANNEL":
+                        return tmpStatus;
+                    case "CH_FAILED INVALID_CHANNEL":
+                        return tmpStatus;
```

Comment:
You should take advantage of switch statement fall-throughs when possible.
```suggestion
                    case "CH_FAILED RECORDING":
                    case "CH_FAILED MISSING_CHANNEL":
                    case "CH_FAILED MALFORMED_CHANNEL":
                    case "CH_FAILED INVALID_CHANNEL":
                        return tmpStatus;
```